documentation/concepts/determinism/index.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: sv_wiki
+title: Modular determinism 
+---
+
+## Running the analysis
+
+With a file such as:
+
+```
+grammar determinism
+
+import host:grammar;
+
+copper_mda testGrammar(hostParser) {
+   extension:grammmar;
+}
+
+copper_mda testGrammarTwo(hostParser) {
+   extension:grammmar_two;
+}
+
+... etc 
+```
+
+Run ```silver --clean determinism```
+
+Unlike in well-definedness, determinism tends to report errors quickly. Determinism will usually stop evaluating errors after the first error it reports, as a lot of errors are reported as bugs or crashes in copper (ideally this will change). Determinism will also not evaluate any grammars following one that fails in a list of ```copper_mda``` definitions, and will evaluate those definitions in order from top to bottom.
+
+---
+
+### Types of Errors Reported
+
+1. <... a long stack trace> ... Requested head of nil.
+
+   This is a bug in copper, where it doesn't check the length of a list before obtaining the first element of the list.
```

Comment:
Really this is a bug in Silver, and the explanation should really be that "Silver fails to notice that a necessary grammar was left out of the parser specification."

Likewise below, the solution I'd say something like "Make sure you've included the right set of grammars that add to concrete syntax (don't forget any terminals or lexer classes, if those are separate)"